### DISCUSSION TITLE
Verwendung von Frontend Frameworks

### DISCUSSION THREAD

**Discussion by thelf** [Webmasters-Europe/codemeetup-forum#15] [Ideas]
Hallo zusammen,

ich habe mich gefragt ob wir bei diesem Projekt Frontend Frameworks wie z.B. Bootstrap oder ähnliche verwenden sollten. Vorteile aus meiner Sicht wären, das wir uns wir Arbeit sparen um z.B. ein flexibles Gestaltungsraster in HTML und CSS zu schreiben. Die Elemente aus einem Framework funktionieren in der Regel Browser übergreifend, so dass wir nicht jedes Element Extra testen müssen. Ein Vorteil wäre auch das leichte Updaten wenn wir das Projekt später anpassen da wir Standards verwenden.

Nachteil ist vielleicht  die Dateigröße(n), wobei man hier normalerweise auch Funktionen und Module leicht entfernen kann wenn man sie nicht benötigt. Ebenfalls benötigt man am Anfang Zeit um sich in das Framework einzuarbeiten was sich meiner Meinung nach aber später rechnet da man wieder Zeit einspart.

Ich selbst arbeite schon seit ein paar Jahren mit Bootstrap, ich kenne aber keine neueren Frameworks. Gestern hat Niclas Tailwind erwähnt, ich weiß nicht ob das ein Frontend-Framework ist.

Was mein ihr dazu, kennt ihr Frontend-Framworks, sollten wir welche einsetzen und wenn ja welche?

**Comment by milack68** [comment]
Ich bin auf jeden Fall dafür, dass wir ein CSS Framework verwenden. Alle Styles manuell zu erstellen ist verdammt viel (und eigentlich unnötige) Arbeit. Ich kenne beides: [Bootstrap ](https://getbootstrap.com/) und [Tailwind](https://tailwindcss.com/). Und auch mit [PaperCSS ](https://www.getpapercss.com/)habe ich eine Einsendeaufgabe erstellt. Das passt aber vom Design her nicht in unser Projekt. Ich habe mir auch schon einmal [Bulma ](https://bulma.io/) angeschaut - dieses Framework wirbt dafür, dass man ohne Javascript auskommt.
In letzter Zeit verwende ich immer öfters Tailwind.  Tailwind ist in der Laravel-Community aktuell sehr beliebt. 
Bootstrap hätte aber den Vorteil, dass es mehr vorgefertigte Komponenten gibt und "der Platzhirsch" unter den Frontend-Frameworks ist. Vermutlich kennen Bootstrap auch mehr von uns, was sicherlich ein Vorteil in der Entwicklung ist.
Aber auch Tailwind ist sehr gut dokumentiert. 
Vielleicht kann Niclas uns noch ein bisserl mehr über die Vor- und Nachteile sagen.

Habe gerade folgenden Link gefunden:
https://www.designerinaction.de/tipps-tricks/web-development/css-frameworks/
Hier werden verschiedene Frameworks miteinander verglichen - sehr interessanter Artikel!

**Reply by alenavershinina** [reply]
Ich finde auch die Verwendung eines Frameworks sinnvoll. Das spart uns Zeit und wir koennen einfach die Seite responsive machen, was jetzt sehr wichtig ist.
Ich habe Erfahrung nur mit Bootstrap. Ueber Tailwind habe ich nicht gehoert.
